### PR TITLE
Add an error message when attempting to transfer from the anonymous principal.

### PR DESCRIPTION
**Pull request by jplevyak**
And provide a hint to use --pem-file <pem-file>

Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

File: src/commands/transfer.rs [hunk 1]
```diff
@@ -32,6 +32,11 @@ pub async fn exec(
     pem: &Option<String>,
     opts: TransferOpts,
 ) -> AnyhowResult<Vec<IngressWithRequestId>> {
+    if pem.is_none() {
+        return Err(anyhow!(
+            "Cannot transfer from anonymous principal, did your forget --pem-file <pem-file> ?"
```
Comment (chmllr) [inline src/commands/transfer.rs:37]:
```suggestion
            "Cannot transfer from anonymous principal, did you forget --pem-file <pem-file> ?"
```

**Review by chmllr — COMMENTED** [review]
Thanks so much John, this is a really annoying problem. But I think we should fix it one level above and then for all commands expecting a pem: https://github.com/dfinity/quill/blob/master/src/commands/mod.rs#L36

So I'd move this check into a short function and call it for every command, whose `exec` function expects a `pem`. WDYT?

File: src/commands/transfer.rs (hunk 1 above)
Comment (jplevyak) [reply src/commands/transfer.rs]: done

**Comment by jplevyak** [comment]
moved up a level and added check for other commands.
PTAL and tell me if I got the check for all the right commands.

File: src/commands/public.rs [hunk 2]
```diff
@@ -37,12 +37,10 @@ fn get_public_ids(
 
 /// Returns the account id and the principal id if the private key was provided.
 pub fn get_ids(pem: &Option<String>) -> AnyhowResult<(Principal, AccountIdentifier)> {
-    let principal_id = get_identity(
-        pem.as_ref()
-            .ok_or_else(|| anyhow!("No PEM file provided"))?,
-    )
-    .sender()
-    .map_err(|e| anyhow!(e))?;
+    require_pem(pem)?;
```
Comment (chmllr) [inline src/commands/public.rs:40]: I think this should go up to the match on commands, right?

**Review by chmllr — APPROVED** [review]
Perfect, thanks so much John!

**Comment by jplevyak** [comment]
I have no right to merge.

**Pull request merged**